### ISSUE TITLE
Setting custom archive and derived data folders

### ISSUE THREAD

**Issue by Panajev** [fastlane/fastlane#1250]
Hello,

I have seen an option for custom archive location, but it was not clearly prominent (although I could understand why since sharing archive folder across lanes/build jobs on a CI server is not the end of the world).

My biggest issue though is that I could not find a command line option to make sure that a custom derived data folder is used for every related (xcodebuild) action. Running multiple parallel build jobs on the same CI server may make sharing the same Derived Data folder problematic, especially if some of those jobs are cleaning it before starting it.

If I try to specify a custom flag for xcodebuild like I currently do for the other projects I have setup Jenkins for, I get an error that states that the custom derived data folder cannot be set twice. The various fastlane tools already do that then, so there should be a way to set it externally as well.

Could you please look into this? Thanks :+1:

**Comment by Panajev** [comment]
@KrauseFx should I open this in a different fastlane tool's repo? I fear it is something trivially simple that I am missing for some reason... :/...

**Comment by Panajev** [comment]
Hello @samrobbins and @KrauseFx , is the -f option in gym the one I would need to reliably make all my builds independent as far as both Derived Data (important for unit testing) and -b for the Archive artefacts (important for Crashlytics/Fabric)? 

`-f, --derived_data_path STRING The directory where build products and other derived data will go (GYM_DERIVED_DATA_PATH)`
`-b, --archive_path STRING The directory in which the archive file should be stored in (GYM_ARCHIVE_PATH)`

I want to make sure I can reasonably clean all previous builds artefact for a project before starting the current operation as well as execute parallel build jobs for the same project on the same slave (a job might test one or more branches and another job might make multiple archives to be shipped to beta testers).

How should I best set it up in my fastfile to make sure this option is respected in all the possible build lanes?

**Comment by vpolouchkine** [comment]
It looks like there hasn’t been any recent activity on this issue in some time, so I’m closing it. If this is still a problem, feel free to re-open a new issue and I’d be happy to help :+1:
